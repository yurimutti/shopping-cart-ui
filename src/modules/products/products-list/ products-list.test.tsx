import { render, screen, fireEvent } from "@testing-library/react";
import { describe, it, expect, vi } from "vitest";
import { ProductsList } from "./";
import type { Product as ProductType } from "@/types/product";

const mockProducts: ProductType[] = [
  {
    id: 1,
    name: "Cheeseburger Combo",
    price: 12.99,
    quantity: 1,
  },
  {
    id: 2,
    name: "Large Pepperoni Pizza",
    price: 18.99,
    quantity: 1,
  },
];

describe("<ProductsList />", () => {
  it("should render all products", () => {
    const { container } = render(
      <ProductsList products={mockProducts} onAddToCart={vi.fn()} />
    );

    mockProducts.forEach((product) => {
      expect(screen.getByText(product.name)).toBeInTheDocument();
      expect(
        screen.getByRole("button", {
          name: `Add ${product.name} to cart`,
        })
      ).toBeInTheDocument();
    });

    expect(container.firstChild).toMatchSnapshot();
  });

  it("should call onAddToCart when add button is clicked", () => {
    const onAddToCart = vi.fn();

    render(<ProductsList products={mockProducts} onAddToCart={onAddToCart} />);

    const addButton = screen.getByRole("button", {
      name: `Add ${mockProducts[0].name} to cart`,
    });

    fireEvent.click(addButton);

    expect(onAddToCart).toHaveBeenCalledWith(mockProducts[0]);
  });
});
