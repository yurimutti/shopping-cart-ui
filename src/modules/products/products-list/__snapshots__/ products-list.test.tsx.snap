// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<ProductsList /> > should render all products 1`] = `
<div
  class="grid grid-cols-1 sm:grid-cols-2 md:grid-cols-3 lg:grid-cols-4 xl:grid-cols-5 gap-6 w-full max-w-7xl mb-12 mx-auto px-0 sm:px-4"
>
  <div
    class="rounded-2xl bg-white p-6 flex flex-col items-center shadow-md border border-gray-200"
  >
    <h2
      class="text-lg font-semibold mb-2 text-black text-center"
    >
      Cheeseburger Combo
    </h2>
    <p
      class="text-gray-700 mb-4"
    >
      $
      12.99
    </p>
    <button
      aria-label="Add Cheeseburger Combo to cart"
      class="rounded-full bg-red-600 hover:bg-red-700 text-white px-4 py-2 text-sm font-medium transition-colors"
    >
      Add to Cart
    </button>
  </div>
  <div
    class="rounded-2xl bg-white p-6 flex flex-col items-center shadow-md border border-gray-200"
  >
    <h2
      class="text-lg font-semibold mb-2 text-black text-center"
    >
      Large Pepperoni Pizza
    </h2>
    <p
      class="text-gray-700 mb-4"
    >
      $
      18.99
    </p>
    <button
      aria-label="Add Large Pepperoni Pizza to cart"
      class="rounded-full bg-red-600 hover:bg-red-700 text-white px-4 py-2 text-sm font-medium transition-colors"
    >
      Add to Cart
    </button>
  </div>
</div>
`;
